### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by friism**
**- What I did**

`make build` doesn't work, so I fixed it. There were two problems:

 * `source` is not in `sh` on Ubuntu: http://stackoverflow.com/a/13702876/2942
 * `CGO_ENABLED` was not set in the scripts invoked from the makefile

**- How to verify it**

`make build`

### CODE REVIEW

**Comment by dnephin** [comment]
I believe this is also fixed by #78 which includes some other stuff for cross.

File: scripts/build/binary [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env sh
 
-source ./scripts/build/ldflags
+. ./scripts/build/ldflags
 
-go build -o ./build/docker --ldflags "${LDFLAGS}" github.com/docker/cli/cmd/docker
+CGO_ENABLED=0 go build -o ./build/docker --ldflags "${LDFLAGS}" github.com/docker/cli/cmd/docker
```
Comment (dnephin) [inline scripts/build/binary]: I think CGO_ENABLED is already set in the Dockerfiles, no?

Comment (friism) [inline scripts/build/binary]: Yes, but these scripts are invoked from `make` outside the Dockerfiles too so builds not using Docker won't work without this setting.

**Comment by friism** [comment]
> I believe this is also fixed by #78 which includes some other stuff for cross.

@dnephin that PR adds `bash` as a requirement which means the build won't work in builds in standard alpine-based containers. Using bash in the shebang was also my first impulse, but I found that caused the Docker-based build to fail.

**Comment by dnephin** [comment]
Sure, it's also going to fail in any container that doesn't have go, or sh. Why would it run in any container but the one we've defined in this repo?  My change adds  bash to the Dockerfile.

File: scripts/build/binary (hunk 1 above)
Comment (dnephin) [inline scripts/build/binary]: True, this is also fixed in #78

**Comment by friism** [comment]
It would run outside of a container (in my case on Ubuntu). That's the primary documented way to build the project: https://github.com/docker/cli#build-locally

Solving the problem by removing dependencies and complexity rather than piling more on top seems preferable.

File: scripts/build/binary (hunk 1 above)
Comment (friism) [inline scripts/build/binary]: Cool @dnephin do you want me to remove that change from this pr?

**Comment by dnephin** [comment]
`bash` exists on ubuntu, so why would it fail when using bash? 

We should also fix those instructions. I don't think running outside a container should be recommended.

**Comment by friism** [comment]
@dnephin it wouldn't fail on Ubuntu when using bash, but you didn't answer my question: Why solve the problem by adding more complexity rather than removing complexity?

**Comment by dnephin** [comment]
You didn't answer my question either! You said it would fail on `alpine` images, and I was asking why we would support arbitrary images!

I agree we should keep dependencies to a minimum. I don't think bash is an unreasonable dependency.  We need to use bash to support `-o pipefail` anyway, so it's not an optional dependency. 

I don't see either option are more complex. Where is the additional complexity?

**Pull request closed without merging**